hadoop-hdfs-project/hadoop-hdfs-httpfs/src/main/java/org/apache/hadoop/fs/http/server/FSOperations.java

```diff
@@ -2300,4 +2316,38 @@ public Map execute(FileSystem fs) throws IOException {
       return toJson(status);
     }
   }
+
+  /**
+   * Executor that performs a fs-status FileSystemAccess files
+   * system operation.
+   */
+  @InterfaceAudience.Private
+  public static class FSStatus
+      implements FileSystemAccess.FileSystemExecutor<Map> {
+    final private Path path;
+
+    /**
+     * Creates a fs-status executor.
+     *
+     * @param path the path to retrieve the status.
+     */
+    public FSStatus(String path) {
+      this.path = new Path(path);
+    }
+
+    /**
+     * Executes the filesystem getStatus operation and returns the
```

Comment:
Notes are repeated a lot from the previous ones, can we have a simpler way to write them?

Reply:
```
/**
     * Executes the filesystem getFileLinkStatus operation and returns the
     * result in a JSON Map.
     *
     * @param fs filesystem instance to use.
     * @return a Map object (JSON friendly) with the file status.
     * @throws IOException thrown if an IO error occurred.
     */
    @Override
    public Map execute(FileSystem fs) throws IOException {
      FileStatus status = fs.getFileLinkStatus(path);
      HttpFSServerWebApp.get().getMetrics().incrOpsStat();
      return toJson(status);
    }
```